### PR TITLE
Add comprehensive unit tests for selector, functions, and helpers

### PR DESCRIPTION
**Pull request by D21IT185BapodraRajSatish**
- Implement tests for various selector functionalities including index, pipe, and dimension selection.
- Add tests for functions like SumFunc, AvgFunc, MinFunc, MaxFunc, CountFunc, ConcatFunc, FirstFunc, and LastFunc.
- Include tests for helper methods such as ValueOf, AsType, and AsArray.
- Ensure edge cases and error handling are covered in the test scenarios.

### CODE REVIEW

File: functions_test.go [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+func TestSumFuncWithInvalidInput(t *testing.T) {
+	query := &Query{}
+	current := Map{}
+	functionOptions := &FunctionOptions{}
+	args := []any{"not a list"}
+
+	_, err := SumFunc(query, current, functionOptions, args)
+
+	if err == nil {
```
[the remark this answers is not in the document]
Comment (D21IT185BapodraRajSatish) [reply functions_test.go:34]: @Vedadiyan  The error message assertion has been added

File: functions_test.go [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+		t.Fatalf("expected no error, got %v", err)
+	}
+
+	expected := 1.9
+	if result != expected {
+		t.Errorf("expected %v, got %v", expected, result)
+	}
+}
+
+func TestMinFuncHandlesMixedNumericTypes(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (D21IT185BapodraRajSatish) [reply functions_test.go:93]: @Vedadiyan Veda The duplicate test case has been removed
